src/core.c/CompUnit/Repository/Installation.rakumod

```diff
@@ -366,6 +382,62 @@ sub MAIN(*@, *%) {
         }
     } ) }
 
+    method !regenerate-bin-wrappers() {
+        $!lock.protect( {
+
+        my $lock = $.prefix.add('repo.lock').open(:create, :w);
+        $lock.lock;
+        LEAVE .unlock, .close with $lock;
+
+        my $version = self!repository-version;
+        self.upgrade-repository unless $version == 2;
+
+        for self.installed() -> $distribution {
+            my $dist  = CompUnit::Repository::Distribution.new($distribution);
+            my @files = $dist.meta<files>.grep(*.defined).map: -> $link {
+                $link ~~ Str ?? $link !! $link.keys[0]
```

Comment:
I think it would be nice if the logic for this loop was reused in the staging repository since this `$link ~~ Str ?? $link !! $link.keys[0]` logic is a code smell (that I wrote 😬) and it would be best to isolate it as much as possible

Reply:
Done. I've contained in a method in the `CompUnit::Repository::Distribution` class.